accession-commons-mongodb/src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/mongodb/service/BasicMongoDbInactiveAccessionServiceTest.java

```diff
@@ -0,0 +1,161 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.commons.accession.persistence.mongodb.service;
+
+import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
+import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
+import com.lordofthejars.nosqlunit.mongodb.MongoDbConfigurationBuilder;
+import com.lordofthejars.nosqlunit.mongodb.MongoDbRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.commons.accession.core.OperationType;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.IOperation;
+import uk.ac.ebi.ampt2d.test.configuration.MongoDbTestConfiguration;
+import uk.ac.ebi.ampt2d.test.persistence.document.TestDocument;
+import uk.ac.ebi.ampt2d.test.persistence.document.TestOperationDocument;
+import uk.ac.ebi.ampt2d.test.persistence.repository.TestRepository;
+import uk.ac.ebi.ampt2d.test.persistence.service.TestMongoDbInactiveAccessionService;
+import uk.ac.ebi.ampt2d.test.rule.FixSpringMongoDbRule;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static uk.ac.ebi.ampt2d.commons.accession.core.OperationType.DEPRECATED;
+import static uk.ac.ebi.ampt2d.commons.accession.core.OperationType.MERGED_INTO;
+import static uk.ac.ebi.ampt2d.commons.accession.core.OperationType.UPDATED;
+import static uk.ac.ebi.ampt2d.test.persistence.document.TestDocument.document;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {MongoDbTestConfiguration.class})
+public class BasicMongoDbInactiveAccessionServiceTest {
+
+    private static final String DEFAULT_REASON = "default-test-reason";
+    @Rule
+    public MongoDbRule mongoDbRule = new FixSpringMongoDbRule(MongoDbConfigurationBuilder.mongoDb()
+            .databaseName("accession-test").build());
+
+    //Required for nosql unit
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Autowired
+    private TestRepository repository;
+
+    @Autowired
+    private TestMongoDbInactiveAccessionService service;
+
+    @UsingDataSet(loadStrategy = LoadStrategyEnum.DELETE_ALL)
+    @Test
+    public void testLastOperationDoesNotExistBehaviour() {
+        new LastOperationAsserts("notExist").doesNotExist();
+    }
+
+    @UsingDataSet(loadStrategy = LoadStrategyEnum.DELETE_ALL)
+    @Test
+    public void testUpdate() {
+        update(document(1, "test-update-1")).exists().isUpdate();
```

Comment:
Without the `new LastOperationAsserts` before them, the calls to `exists().isUpdate()` are not obvious in my opinion. Renaming them to `assertOperationExists().assertIsUpdate()` would be more verbose but also completely clear.